src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/SearchParameterOperations.cs

```diff
@@ -176,5 +188,86 @@ public async Task UpdateSearchParameterAsync(ITypedElement searchParam, RawResou
                 throw customSearchException;
             }
         }
+
+        /// <summary>
+        /// This method should be called periodically to get any updates to SearchParameters
+        /// added to the DB by other service instances.
+        /// It should also be called when a user starts a reindex job
+        /// </summary>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns>A task.</returns>
+        public async Task GetAndApplySearchParameterUpdates(CancellationToken cancellationToken)
+        {
+            var updatedSearchParameterStatus = await _searchParameterStatusManager.GetSearchParameterStatusUpdates();
+
+            // first process any deletes, then we will do any adds or updates
+            // this way any deleted params which might have the same code or name as a new
+            // parameter will not cause conflicts
+            foreach (var searchParam in updatedSearchParameterStatus.Where(p => p.Status == SearchParameterStatus.Deleted))
+            {
+                DeleteSearchParameter(searchParam.Uri.ToString());
+            }
+
+            var paramsToAdd = new List<ITypedElement>();
+            foreach (var searchParam in updatedSearchParameterStatus.Where(p => p.Status != SearchParameterStatus.Deleted))
+            {
+                var searchParamResource = await GetSearchParameterByUrl(searchParam.Uri.ToString(), cancellationToken);
+
+                if (searchParamResource == null)
+                {
+                    _logger.LogError(
+                        "Updated SearchParameter status found for SearchParameter: {0}, but did not find any SearchParameter resources when querying for this url.",
+                        searchParam.Uri);
```

Comment:
Perhaps a warning instead?

```suggestion
                    _logger.LogWarning(
                        "Updated SearchParameter status found for SearchParameter: {0}, but did not find any SearchParameter resources when querying for this url.",
                        searchParam.Uri);
```